src/ImageSharp/Processing/Processors/Transforms/Resize/ResamplerExtensions.cs

```diff
@@ -0,0 +1,227 @@
+            using var horizontalKernelMap = ResizeKernelMap<TResampler>.Calculate(
+                in sampler,
+                destinationRectangle.Width,
+                sourceRectangle.Width,
+                allocator);
+
+            using var verticalKernelMap = ResizeKernelMap<TResampler>.Calculate(
+                in sampler,
+                destinationRectangle.Height,
+                sourceRectangle.Height,
+                allocator);
+
+            for (int i = 0; i < source.Frames.Count; i++)
+            {
+                ImageFrame<TPixel> sourceFrame = source.Frames[i];
+                ImageFrame<TPixel> destinationFrame = destination.Frames[i];
+
+                ApplyResizeFrameTransform(
+                    configuration,
+                    sourceFrame,
+                    destinationFrame,
+                    horizontalKernelMap,
+                    verticalKernelMap,
+                    sourceRectangle,
+                    destinationRectangle,
+                    interest,
+                    compand);
+            }
+        }
+
+        private static void ApplyNNResizeFrameTransform<TPixel>(
+            Configuration configuration,
+            ImageFrame<TPixel> source,
+            ImageFrame<TPixel> destination,
+            Rectangle sourceRectangle,
+            Rectangle destinationRectangle,
+            Rectangle interest)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            // Scaling factors
+            float widthFactor = sourceRectangle.Width / (float)destinationRectangle.Width;
+            float heightFactor = sourceRectangle.Height / (float)destinationRectangle.Height;
+
+            var operation = new NNRowIntervalOperation<TPixel>(
+                sourceRectangle,
+                destinationRectangle,
+                widthFactor,
+                heightFactor,
+                source,
+                destination);
+
+            ParallelRowIterator.IterateRows(
+                configuration,
+                interest,
+                in operation);
+        }
+
+        private static void ApplyResizeFrameTransform<TResampler, TPixel>(
+            Configuration configuration,
+            ImageFrame<TPixel> source,
+            ImageFrame<TPixel> destination,
+            ResizeKernelMap<TResampler> horizontalKernelMap,
+            ResizeKernelMap<TResampler> verticalKernelMap,
+            Rectangle sourceRectangle,
+            Rectangle destinationRectangle,
+            Rectangle interest,
+            bool compand)
+            where TResampler : unmanaged, IResampler
+            where TPixel : struct, IPixel<TPixel>
+        {
+            PixelConversionModifiers conversionModifiers =
+                PixelConversionModifiers.Premultiply.ApplyCompanding(compand);
+
+            BufferArea<TPixel> sourceArea = source.PixelBuffer.GetArea(sourceRectangle);
+
+            // To reintroduce parallel processing, we would launch multiple workers
```

Comment:
@antonfirsov I'm wondering why we haven't done this?

Reply:
Shouldn't be rocket science, but still non trivial, had to stop somewhere.

I'm not the biggest fan of parallelization anyways, since it's mostly for desktop and low-load web workflows, not sure if we picked the best defaults for `LevelOfParallelism`.

Reply:
I think desktop is a reasonable target and will be a very popular one once we expand our format support. Web applications tend to have less image processing throughput than you might imagine (sensible ones anyway) since they would employ pre generation and caching for the most part. 

Regarding `LevelOfParalleism`, Yeah I don't know, I guess research and experimentation...

Not a major concern anyway. Our Resize is fast enough for now and focus is better spent elsewhere.